src/Discovery/Archive.hs

```diff
@@ -87,15 +87,15 @@ withArchive' file go = traverse (\e -> withArchive e file go) (selectUnarchiver
 -- on the temporary directory. Archive contents are removed when the callback
 -- finishes.
 withArchive ::
-  (Has (Lift IO) sig m, Has Finally sig m) =>
+  (Has (Lift IO) sig m, Has Diagnostics sig m, Has Finally sig m) =>
   -- | Archive extraction function
   (Path Abs Dir -> Path Abs File -> m ()) ->
   -- | Path to archive
   Path Abs File ->
   -- | Callback
   (Path Abs Dir -> m c) ->
   m c
-withArchive extract file go = do
+withArchive extract file go = fatalOnSomeException "withArchive" $ do
```

Comment:
Without this we can't capture unhandled exception originating from zip (e.g. for corrupted jar)